### PR TITLE
Add graphql client error handling BBE

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/graphql-client-error-handling/graphql_client_error_handling.bal [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import ballerina/graphql;
+import ballerina/io;
+
+type ProfileResponse record {|
+    *graphql:GenericResponseWithErrors;
+    record {|Person profile;|} data;
+|};
+
+type Person record {|
```
Comment (ThisaruGuruge) [inline examples/graphql-client-error-handling/graphql_client_error_handling.bal:9]: Let's use `Profile` as type name in all the places.

```suggestion
type Profile record {|
```

File: examples/graphql-client-error-handling/graphql_client_error_handling.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+## Prerequisites
+- Run the GraphQL service given in the [Mutations](https://ballerina.io/learn/by-example/graphql-mutations/) example.
+
+Run the client program by executing the following command.
+
+::: out graphql_client_error_handlingt.out :::
+
+## Related links
+- [`graphql:ClientError` - API documentation](https://lib.ballerina.io/ballerina/graphql/1.4.4/errors#ClientError)
```
Comment (ThisaruGuruge) [inline examples/graphql-client-error-handling/graphql_client_error_handling.md:15]:
```suggestion
- [`graphql:ClientError` error - API documentation](https://lib.ballerina.io/ballerina/graphql/1.4.4/errors#ClientError)
```

File: examples/graphql-client-error-handling/graphql_client_error_handling.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- [Client error handling - Specification](/spec/graphql/#255-client-error-handling)
```
Comment (ThisaruGuruge) [inline examples/graphql-client-error-handling/graphql_client_error_handling.md:16]:
```suggestion
- [`graphql:Client` error handling - Specification](/spec/graphql/#255-client-error-handling)
```

File: examples/graphql-client-error-handling/graphql_client_error_handling.bal [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        handleErrors(err);
+    }
+}
+
+function handleErrors(graphql:ClientError clientError) {
+    if clientError is graphql:PayloadBindingError {
+        // This error represents a client-side data binding error. This error occurs due to the
+        // assigned variable type and the value obtained from the wire having a mismatching 
+        // shape.
+        io:println("PayloadBindingError: ", clientError.cause());
```
Comment (ThisaruGuruge) [inline examples/graphql-client-error-handling/graphql_client_error_handling.bal:31]: Cause can be `nil`, right? Better to use the error message here.

```suggestion
        io:println("PayloadBindingError: ", clientError.message());
```

File: examples/graphql-client-error-handling/graphql_client_error_handling.bal [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        // This error represents GraphQL errors due to GraphQL server-side document 
+        // validation. The GraphQL errors returned from the server-side can be obtained by 
+        // calling the `detail()` method on `graphql:InvalidDocumentError`.
+        graphql:ErrorDetail[]? errorDetails = clientError.detail().errors;
+        io:println("InvalidDocumentError: ", errorDetails);
+    } else if clientError is graphql:HttpError {
+        // This error represents network-level errors. If the response from the server contains 
+        // a body then, it can be obtained by calling the `detail()` method on `graphql:HttpError`.
+        anydata body = clientError.detail().body;
+        io:println("HttpError: ", body, clientError.cause());
```
Comment (ThisaruGuruge) [inline examples/graphql-client-error-handling/graphql_client_error_handling.bal:42]: Ditto.

File: examples/graphql-client-error-handling/graphql_client_error_handling.bal (hunk 1 above)
Comment (MohamedSabthar) [reply examples/graphql-client-error-handling/graphql_client_error_handling.bal:9]: Updated

File: examples/graphql-client-error-handling/graphql_client_error_handling.bal (hunk 4 above)
Comment (MohamedSabthar) [reply examples/graphql-client-error-handling/graphql_client_error_handling.bal:31]: Updated

File: examples/graphql-client-error-handling/graphql_client_error_handling.md (hunk 2 above)
Comment (MohamedSabthar) [reply examples/graphql-client-error-handling/graphql_client_error_handling.md:15]: Updated

File: examples/graphql-client-error-handling/graphql_client_error_handling.md (hunk 3 above)
Comment (MohamedSabthar) [reply examples/graphql-client-error-handling/graphql_client_error_handling.md:16]: Updated

File: examples/graphql-client-error-handling/graphql_client_error_handling.bal (hunk 5 above)
Comment (MohamedSabthar) [reply examples/graphql-client-error-handling/graphql_client_error_handling.bal:42]: Updated

**Pull request merged**